Common/Orders/Fees/GDAXFeeModel.cs

```diff
@@ -40,21 +47,73 @@ public override OrderFee GetOrderFee(OrderFeeParameters parameters)
             var security = parameters.Security;
 
             // marketable limit orders are considered takers
-            decimal fee = 0;
-            // check limit order posted to the order book, 0% maker fee
-            if (!(order.Type == OrderType.Limit && !order.IsMarketable))
+            var isMaker = order.Type == OrderType.Limit && !order.IsMarketable;
+
+            var feePercentage = GetFeePercentage(order.Time, isMaker);
+
+            // get order value in quote currency, then apply maker/taker fee factor
+            var unitPrice = order.Direction == OrderDirection.Buy ? security.AskPrice : security.BidPrice;
+            unitPrice *= security.SymbolProperties.ContractMultiplier;
+
+            // currently we do not model 30-day volume, so we use the first tier
+
+            var fee = unitPrice * order.AbsoluteQuantity * feePercentage;
+
+            return new OrderFee(new CashAmount(fee, security.QuoteCurrency.Symbol));
+        }
+
+        /// <summary>
+        /// Returns the maker/taker fee percentage effective at the requested date.
+        /// </summary>
+        /// <param name="utcTime">The date/time requested (UTC)</param>
+        /// <param name="isMaker">true if the maker percentage fee is requested, false otherwise</param>
+        /// <returns>The fee percentage effective at the requested date</returns>
+        public static decimal GetFeePercentage(DateTime utcTime, bool isMaker)
+        {
+            for (var index = FeeHistory.Count - 1; index >= 0; index--)
             {
-                // get order value in quote currency, then apply taker fee factor
-                var unitPrice = order.Direction == OrderDirection.Buy ? security.AskPrice : security.BidPrice;
-                unitPrice *= security.SymbolProperties.ContractMultiplier;
+                var entry = FeeHistory[index];
```

Comment:
Let's keep this simpler for now; we can add your style back if they change fees a third time:
```
var fees = [0, 0.03];
if (date > 2019,03,23) fees = [0.015, 0.03]
```